.github/workflows/ci.yml

```
@@ -8,6 +8,14 @@ on:
   schedule:
     - cron: 0 0 * * *
 
+# Force cancellation of in-progress workflows if changes are made to
+# the HEAD of the branch the workflow is currently running on.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  # master branch will be allowed to have pending jobs
+  # https://stackoverflow.com/questions/68418857/how-to-cancel-existing-runs-when-a-new-push-happens-on-github-actions-but-only
```

```suggestion
  # https://stackoverflow.com/a/70972844
```

The permalink to the specific answer is a bit shorter and I think more likely to not get broken